File: 27-remove-element/remove-element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        if(nums.size()==0)
            return 0;
        if(nums.size()==1 && nums[0]==val)
            return 0;
        if(nums.size()==1 && nums[0]!=val)
            return 1;
        int i,j,count=0;
        for(i=0;i<nums.size()-count;i++){
            if(nums[i]==val){
                count++;
                j=i;
                while(j<nums.size()-1){
                    nums[j]=nums[j+1];
                    j++;
                }
                i--;
                if(i+1==nums.size()-count)
                    return i+1;
            }
        }
        return i;      
    }
};

File: 27-remove-element/remove-element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        if(nums.length==0)
            return 0;
        if(nums.length==1 && nums[0]==val)
            return 0;
        if(nums.length==1 && nums[0]!=val)
            return 1;
        int i,j,temp;
        for(i=0,j=nums.length-1;i<=j;i++){
            if(nums[j]==val){
                j--;
                i--;
                continue;
            }
            if(nums[i]==val){
                temp=nums[i];
                nums[i]=nums[j];
                nums[j]=nums[i];
                j--;
            }
        }
        return j+1;
    }
}